File: Practice/codewars/1.DetectPanagram/js/solution.js
function format(string) {
  let res = "";

  string.toLowerCase() //to lowercase
        .replace(/\s/g, "") //delete whitespaces
        .split("") // make string array
        .forEach(ch => { //foreach i in array
    if (res.indexOf(ch) === -1) {
      res += ch
    }
  })

  return res
}

function isPangram(string){
  let alphabet = "abcdefghijklmnopqrstuvwxyz";
  let counter = 0;

  format(string).split("").forEach(ch => {
    if (alphabet.indexOf(ch) !== -1) {
      counter++;
    }
  });

  if (counter >= alphabet.length) {
    return true;
  } else {
    return false;
  }
}

File: Practice/codewars/1.DetectPanagram/python/solution.py
import string


def format(s):
    res = ""
    for ch in s.lower().replace(" ", ""):
        if ch not in res:
            res = res + ch
    return res


def is_pangram(s):
    alphabet = list(string.ascii_lowercase)
    counter = 0

    for l in format(s):
        if l in alphabet:
            counter += 1

    if counter >= len(alphabet):
        return True
    else:
        return False

# pangram = "Aacdefghijklmnopqrstuvwxyz is not a pangram."
# Tricky Case: Repeated Chars, Whitespaces, Uppercase, More than alphabet
# test.assert_equals(is_pangram(pangram), False)
